src/ThreadsCommon.c

```diff
@@ -0,0 +1,160 @@
+#include <ThreadsCommon.h>
+#include <interfaces_info.h>
+#include <rtems.h>
+
+#ifndef RT_MAX_CYCLIC_INTERFACES
+#define RT_MAX_CYCLIC_INTERFACES RUNTIME_THREAD_COUNT
+#endif
+
+#define NANOSECOND_IN_MILISECOND 1000000
+
+extern rtems_name generate_new_partition_timer_name();
+static void schedule_next_tick(const rtems_id timer_id);
+static void timer_callback(rtems_id timer_id, void *unused);
+static void update_execution_time_data(const uint32_t thread_id,
+				       const uint64_t thread_execution_time);
+
+typedef void (*call_function)(const char *buf, size_t len);
+
+struct CyclicRequestData {
+	rtems_id timer_id;
+	rtems_interval next_wakeup_ticks;
+	rtems_interval interval_ticks;
+	uint32_t queue_id;
+	uint32_t request_size;
+	bool is_used;
+};
+
+struct CyclicEmptyRequestData {
+	uint32_t m_sender_pid;
+	uint32_t m_length;
+	uint8_t *m_data __attribute__((aligned(16)));
+};
+
+static struct CyclicRequestData cyclic_reqeust_data[RT_MAX_CYCLIC_INTERFACES];
+static struct CyclicEmptyRequestData empty_request;
+
+static void schedule_next_tick(const rtems_id timer_id)
+{
+	uint32_t index = 0;
+	for (uint32_t index = 0; index < RT_MAX_CYCLIC_INTERFACES; index++) {
+		if (!cyclic_reqeust_data[index].is_used &&
+		    cyclic_reqeust_data[index].timer_id == timer_id) {
+			break;
+		}
+	}
+
+	cyclic_reqeust_data[index].next_wakeup_ticks +=
+	    cyclic_reqeust_data[index].interval_ticks;
+	rtems_interval delta = cyclic_reqeust_data[index].next_wakeup_ticks -
+			       rtems_clock_get_ticks_since_boot();
+	rtems_timer_fire_after(timer_id, delta, timer_callback, NULL);
+}
+
+static void timer_callback(rtems_id timer_id, void *unused)
+{
+	uint32_t index = 0;
+	for (uint32_t index = 0; index < RT_MAX_CYCLIC_INTERFACES; index++) {
+		if (!cyclic_reqeust_data[index].is_used &&
+		    cyclic_reqeust_data[index].timer_id == timer_id) {
+			break;
+		}
+	}
+
+	rtems_message_queue_send(cyclic_reqeust_data[index].queue_id,
+				 &empty_request,
+				 cyclic_reqeust_data[index].request_size);
+	schedule_next_tick(cyclic_reqeust_data[index].timer_id);
+}
+
+static void update_execution_time_data(const uint32_t thread_id,
+				       const uint64_t thread_execution_time)
+{
+
+	if (thread_execution_time <
+	    threads_info[thread_id].min_thread_execution_time) {
+		threads_info[thread_id].min_thread_execution_time =
+		    thread_execution_time;
+	}
+
+	if (thread_execution_time >
+	    threads_info[thread_id].max_thread_execution_time) {
+		threads_info[thread_id].max_thread_execution_time =
+		    thread_execution_time;
+	}
+
+	threads_info[thread_id].mean_thread_execution_time =
+	    threads_info[thread_id].mean_thread_execution_time +
+	    ((double)thread_execution_time -
+	     threads_info[thread_id].mean_thread_execution_time) /
+		((double)threads_info[thread_id].execution_time_counter + 1);
+
+	threads_info[thread_id].execution_time_counter++;
+}
+
+bool ThreadsCommon_CreateCyclicRequest(uint64_t interval_ns, uint32_t queue_id,
```

Comment:
I see that this is consistent with the signature in SDD that I prepared. And that signature is consistent with the logic in the RTEMS runtime. However, now, as I review it, it seems that I missed one thing that is absent from the current RTEMS runtime, but should be included -> offset. See:

```
    taste::Timer::run(@_Dispatch_Offset_ms_@ms, @_TRIM:Period_@ms, [] () -> void
    {
        static taste::Request<@_UPPER:Thread_Name_@_REQUEST_SIZE> request;
        call_@_LOWER:Thread_Name_@((const char*)request.data(), request.length());
    });

```

https://gitlab.esa.int/taste/kazoo/-/blob/feature_debian12/templates/concurrency_view/generic_linux_thread_body/thread.tmplt?ref_type=heads



[please change the signature here and in SDD]